### PR TITLE
Remove a redundant withParameters call in draw-and-pick

### PR DESCRIPTION
**Pull request by shaojingli**
@Pessimistress Could you try this branch to see if the performance regression is still there?
#769

### CODE REVIEW

File: src/lib/draw-and-pick.js [hunk 1]
```diff
@@ -338,34 +341,20 @@ function getPickedColors(gl, {
...
-
-    withParameters(gl, settings, () => {
-      // Render all pickable layers in picking colors
-      layers.forEach((layer, layerIndex) => {
-        if (!layer.isComposite && layer.props.visible && layer.props.pickable) {
-
-          // Encode layerIndex with alpha
-          setParameters(gl, {blendColor: [0, 0, 0, (layerIndex + 1) / 255]});
-          layer.drawLayer({
-            moduleParameters: Object.assign({}, layer.props, {
```
Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: Why remove?

File: src/lib/draw-and-pick.js [hunk 2]
```diff
@@ -338,34 +341,20 @@ function getPickedColors(gl, {
...
-              viewport: layer.context.viewport
-            }),
-            uniforms: Object.assign(
-              {renderPickingBuffer: 1, pickingEnabled: 1},
-              layer.context.uniforms,
-              {layerIndex}
-            ),
-            settings: layer.props.settings || {}
```
Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: And this?

File: src/lib/draw-and-pick.js [hunk 3]
```diff
@@ -338,34 +341,20 @@ function getPickedColors(gl, {
...
+    layers.forEach((layer, layerIndex) => {
+      if (!layer.isComposite && layer.props.visible && layer.props.pickable) {
+
+        // Encode layerIndex with alpha
+        setParameters(gl, {blendColor: [0, 0, 0, (layerIndex + 1) / 255]});
+        layer.drawLayer({
+          uniforms: Object.assign(
+            {renderPickingBuffer: 1, pickingEnabled: 1},
+            layer.context.uniforms,
+            getUniformsFromViewport(layer.context.viewport, layer.props),
```
Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: This function has been removed.

File: src/lib/draw-and-pick.js (hunk 1 above)
Comment (shaojingli) [inline src/lib/draw-and-pick.js]: the settings used in this withParameters() has been accommodated by the outer withParameters() call at line 328

Comment (Pessimistress) [inline src/lib/draw-and-pick.js]: I was referring to moduleParameters, but it seems like a bad merge before.

Comment (ibgreen) [inline src/lib/draw-and-pick.js]: FWIW, moduleParameters is part of the new shader module system

**Pull request merged**